lib/puppet/provider/domain_name/cisco_nexus.rb

```diff
@@ -0,0 +1,25 @@
+require 'puppet/resource_api/simple_provider'
+require 'cisco_node_utils'
+
+class Puppet::Provider::DomainName::CiscoNexus <  Puppet::ResourceApi::SimpleProvider
```

Comment:
```Error: Could not autoload puppet/type/domain_name: Could not autoload puppet/provider/domain_name/cisco_nexus: uninitialized constant Puppet::Provider::DomainName
Did you mean?  DomainName```

Which indicates it is expecting `class DomainName::CiscoName` which is weird. Maybe a provider name clash issue?